osu.Server.Spectator/Hubs/MultiplayerQueue.cs

```diff
@@ -77,17 +81,16 @@ public async Task FinishCurrentItem()
             using (var db = dbFactory.GetInstance())
             {
                 // Expire and let clients know that the current item has finished.
-                await db.ExpirePlaylistItemAsync(CurrentItem.ID);
+                await db.MarkPlaylistItemAsPlayedAsync(CurrentItem.ID);
                 CurrentItem.Expired = true;
+                CurrentItem.PlayedAt = DateTimeOffset.Now;
```

Comment:
Dunno if this is me being paranoid, but I find the fact that both the DB and the server generate this timestamp independently (via SQL `NOW()` and via `DateTimeOffset.Now` respectively) kinda sketchy... I would hope this property is display-only, but if this is to be used in actual logic then that may become an issue.

Reply:
I am tending to agree. We should do a re-query here...